### PR TITLE
Interpret adducts

### PR DESCRIPTION
**Pull request by niekdejonge**
Added functionality to interpret adducts if they are not in known adducts list.

### CODE REVIEW

File: matchms/filtering/repair_adduct/repair_adduct_based_on_smiles.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    adducts_dict = load_adducts_dict()
+    if ion_mode not in ("positive", "negative"):
+        logger.warning(f"Ionmode: {ion_mode} not positive or negative, first run derive_ionmode")
+        return spectrum
+    if precursor_mz is None:
+        logger.warning("Precursor_mz is None, first run add_precursor_mz")
+        return spectrum
+
+    smiles_mass = get_monoisotopic_neutral_mass(spectrum.get("smiles"))
+    for adduct_name in adducts_dict:
```
Comment (niekdejonge) [inline matchms/filtering/repair_adduct/repair_adduct_based_on_smiles.py]: Replace with pandas or numpy. Select the smallest difference.

File: matchms/filtering/repair_parent_mass_from_smiles/repair_parent_mass_is_mol_wt.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import logging
```
Comment (niekdejonge) [inline matchms/filtering/repair_parent_mass_from_smiles/repair_parent_mass_is_mol_wt.py:1]: Add a combination function that repairs parent mass from smiles.

File: setup.py [hunk 3]
```diff
@@ -58,6 +58,7 @@
         "scipy",
         "sparsestack >= 0.4.1",
         "tqdm",
+        "molmass",
```
Comment (niekdejonge) [inline setup.py]: Check if conda package, otherwise use rdkit

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,9 +8,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### Added
+- New filter functions to repair a smiles that mismatches parent mass
```
Comment (florian-huber) [inline CHANGELOG.md]: Expand a bit. 
- Mention PR --> ` [#440](https://github.com/matchms/matchms/pull/440)`
- Make it multiple points to name all things that now can be done (correcting parent mass, precursor mz, adducts...) but also to remove spectra with inconsistencies between precursor and annotated smiles.

**Review by florian-huber — APPROVED** [review]
@niekdejonge Nice additions and good tests!
Apart from the points we just discussed (and a small comment I added to the changelog) this all looks good to me!

**Comment by hechth** [comment]
@niekdejonge I will add my review tomorrow after the call when I know where this is heading :D

**Pull request merged**